### ISSUE TITLE
fixes the missing type annotations

### ISSUE THREAD

**Issue by wyli** [Project-MONAI/MONAI#5605]
**Is your feature request related to a problem? Please describe.**
running the mypy commands
```bash
# python -m pip install -U -r requirements-dev.txt
./runtests.sh --mypy
```
generates the following warning messages, would be great to add type annotations to those functions.

```
monai/data/meta_obj.py:83:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/meta_obj.py:84:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/meta_obj.py:85:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/meta_obj.py:86:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/meta_tensor.py:372:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/transforms/intensity/dictionary.py:1589:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/transforms/intensity/dictionary.py:1662:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/image_reader.py:292:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/image_reader.py:293:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/image_reader.py:613:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/image_reader.py:614:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/image_reader.py:944:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/image_reader.py:945:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/image_reader.py:1110:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/image_reader.py:1111:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/image_reader.py:1201:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/image_reader.py:1202:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
monai/data/dataset.py:1076:13: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
...
```

The warning is because functions that do not have any annotations (neither for any argument nor for the return type) are not type-checked

**Comment by Shadow-Devil** [comment]
Hi, I would like to work on this issue.

**Comment by Shadow-Devil** [comment]
@wyli I went through the code and fixed the MyPy warnings. 
Some notes I would like your comment on:
1. I use a virtual environment where all the dependencies are placed into the `venv/` folder. I've excluded this folder from mypy since otherwise it would find warnings in the dependencies. If you don't use a virtual environment, the exclusion should have no effect.
2. In some cases I had to change the implementation, since e.g. [a variable is reassigned with a different type](https://github.com/Project-MONAI/MONAI/blob/e8c624769b30f55f265995e1c4ff899ead7ff19e/monai/apps/deepgrow/transforms.py#L860-L861).
3. MyPy is still not checking all functions, only those that have type annotations. If you set the config `check_untyped_defs = True` there are still over 1000 warnings and errors. I would also like to tackle this task, should this be done in a separate issue or still be tracked in this?
